stko/molecular/conversion/z_matrix.py

```diff
@@ -0,0 +1,173 @@
+                    atom2_id=mol_atoms[i-1].get_id(),
+                )
+                angle = calculate_angle(
+                    pt1=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-3].get_id()
+                        )
+                    )[0],
+                    pt2=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-2].get_id()
+                        )
+                    )[0],
+                    pt3=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-1].get_id()
+                        )
+                    )[0],
+                )
+
+                distance = round(distance, 2)
+                angle = round(angle, 2)
+                zmatrix.append(
+                    f'{atom.__class__.__name__} {i} {distance}'
+                    f'{i-1} {angle}'
+                )
+                coords += 2
+
+            else:
+                distance = get_atom_distance(
+                    position_matrix=position_matrix,
+                    atom1_id=atom.get_id(),
+                    atom2_id=mol_atoms[i-1].get_id(),
+                )
+                angle = calculate_angle(
+                    pt1=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-3].get_id()
+                        )
+                    )[0],
+                    pt2=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-2].get_id()
+                        )
+                    )[0],
+                    pt3=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-1].get_id()
+                        )
+                    )[0],
+                )
+                torsion = calculate_dihedral(
+                    pt1=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-3].get_id()
+                        )
+                    )[0],
+                    pt2=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-2].get_id()
+                        )
+                    )[0],
+                    pt3=tuple(
+                        molecule.get_atomic_positions(
+                            mol_atoms[i-1].get_id()
+                        )
+                    )[0],
+                    pt4=tuple(
+                        molecule.get_atomic_positions(atom.get_id())
+                    )[0],
+                )
+
+                distance = round(distance, 2)
+                angle = round(angle, 2)
+                torsion = round(torsion, 2)
+                zmatrix.append(
+                    f'{atom.__class__.__name__} {i} {distance}'
+                    f'{i-1} {angle} {i-2} {torsion}'
+                )
+                coords += 3
```

Comment:
Just wanted to check, is it 100% certain that the atom ordering returned from `get_atom_positions()` is systematic and not random?

Reply:
do you mean `get_position_matrix()`? Because if so, yes. In that case, it returns the entire matrix, which is ordered by atom id.